### PR TITLE
Add some light SemVer functionality

### PR DESCRIPTION
**Pull request by Herbstein**

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Let me know when you want some feedback. I note that the build is failing.

**Comment by Herbstein** [comment]
I think this is ready. I'm not entirely happy with the ad-hoc parser-combinator implementation to support the parsing. It was cannibalized from a failed experiment and I didn't want to spend more than than necessary for this small part.

File: main/src/library/Pkger/SemVer.flix [hunk 1]
```diff
@@ -0,0 +1,92 @@
+namespace Pkger/SemVer {
```
Comment (magnus-madsen) [inline main/src/library/Pkger/SemVer.flix:1]: Add copyright headers.

File: main/src/library/Pkger/SemVer.flix [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    ///
+    /// Represents an instance of a Semantic Versioning (SemVer) version.
+    /// It is currently missing pre-release identifiers and build metadata
+    ///
+    pub enum SemVer {
+        case SemVer(Int, Int, Int)
+    }
+
```
Comment (magnus-madsen) [inline main/src/library/Pkger/SemVer.flix:9]: Extra new lines

File: main/src/library/Pkger/SemVer.flix [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                    minor
+                }
+            } else {
+                major
+            }
+        }
+    }
+
+    ///
+    /// Check whether the `specific` SemVer is compatible with the `bound` SemVer
```
Comment (magnus-madsen) [inline main/src/library/Pkger/SemVer.flix:31]: What does it mean to be compatible?

File: main/src/library/Pkger/SemVer.flix [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            Option.flatMap(match (inputx, x) -> runParser(f(x), inputx), runParser(v, input))
+        }
+
+        def return(x: a): Parser[a] = input -> Some((input, x))
+
+        def empty(): Parser[a] = _ -> None
+
+        def char(x: Char): Parser[Char] = input -> match String.toList(input) {
+            case Cons(y, ys) => if (y == x) { Some((List.toString(ys), x)) } else { None }
+            case Nil => None
```
Comment (magnus-madsen) [inline main/src/library/Pkger/SemVer.flix:57]: Base cases are usually first.

File: main/src/library/Pkger/SemVer.flix (hunk 3 above)
Comment (Herbstein) [reply main/src/library/Pkger/SemVer.flix:31]: According to the SemVer specification. Added that to the documentation.

**Pull request merged**